apps/indexer/lib/indexer/buffered_task.ex

```diff
@@ -420,6 +425,19 @@ defmodule Indexer.BufferedTask do
     %BufferedTask{state | bound_queue: new_bound_queue}
   end
 
+  defp dedup_entries(%BufferedTask{dedup_entries: false}, entries), do: entries
+
+  defp dedup_entries(
+         %BufferedTask{dedup_entries: true, task_ref_to_batch: task_ref_to_batch, bound_queue: bound_queue},
+         entries
+       ) do
+    running_entries = Enum.uniq(List.flatten(Map.values(task_ref_to_batch)))
```

Comment:
typically you'd avoid nesting calls like this in elixir and instead pipe values through a pipeline

e.g.

```elixir
running _entries = task_ref_to_batch
  |> Map.values()
  |> List.flatten()
  |> Enum.uniq()
```
they are effectively identical constructs, but the pipe is more idiomatic. 

However in this case you can use a `MapSet` to perform the dedup. This works better as calling `Enum.member?` on a list means the entire list must be traversed each time, whist set membership runs in constant time. You could then do a set intersection operation to find unique values, returning an empty list should there be none.

this would be something like ( i have not tested this)

```elixir
running_entries = task_ref_to_batch
  |> Map.values()
  |> List.flatten()
  |> MapSet.new()

queued_entries = MapSet.new(bound_queue)

entries 
  |> MapSet.new()
  |> MapSet.difference(running_entries)
  |> MapSet.difference(queued_entries)
  |> MapSet.to_list()
```

Reply:
Great point, updated.